corehq/blobs/pillow.py

```diff
@@ -19,46 +14,28 @@
 
 class BlobDeletionProcessor(PillowProcessor):
 
-    def __init__(self, blob_db, bucket_base):
+    def __init__(self, blob_db):
         super(BlobDeletionProcessor, self).__init__()
         self.blob_db = blob_db
-        self.bucket_base = bucket_base
 
     def process_change(self, pillow_instance, change):
         if change.deleted:
-            bucket = join(self.bucket_base, change.id)
+            bucket = "{}/{}".format(change.meta.data_source_name, change.id)
```

Comment:
> I think this PR changes the last thing to:
> * blobs are deleted on hard and soft deletions...
>
> which is probably a net worse change.

Completely agree with this, which is why I added the *hold off on merge* tag (was not planning to remove it until that was resolved).

I think your proposed solutions sound good. +1